.github/workflows/java-publish.yml

```diff
@@ -1,6 +1,8 @@
 name: Build and publish Java packages
 on:
   release:
+    # Use released instead of published, since we don't publish preview/beta
+    # versions
```

Comment:
Do you want to start publishing them later? Feel like it could be useful since we distribute the binaries for Java, right? The only reason we skip Rust is because it's a source-only release, so there's no benefit of publishing over just letting people reference the git repo directly.